e2e/metrics/metrics.go

```diff
@@ -89,7 +89,28 @@ func (tc *MetricsTest) TestMetricsLinux(f *framework.F) {
 	tc.queryAllocMetrics(t, workloads)
 }
 
-// Run workloads from and wait for allocations
+// TestMetricsWindows runs a collection of jobs that exercise alloc metrics.
+// Then we query prometheus to verify we're collecting client and alloc metrics
+// and correctly presenting them to the prometheus scraper.
+func (tc *MetricsTest) TestMetricsWindows(f *framework.F) {
+	t := f.T()
+	clientNodes, err := e2eutil.ListWindowsClientNodes(tc.Nomad())
+	require.Nil(t, err)
+	if len(clientNodes) == 0 {
+		t.Skip("no Windows clients")
```

Comment:
Should we check if we expect windows clients as well?  To avoid accidentally skipping tests in CI if windows clients unexpectedly fail to join cluster?

We ran into such case before where java driver tests were accidentally ignored.  Sadly, java driver tests and `javaCompatible` still doesn't protect against this problem.

Reply:
I think that's a good idea but maybe not at the level of an individual test case. There are plenty of cases where you'll want to stand up a less-than-complete environment and run a subset of tests against it. There's some code in `framework/provisioning` that we're not currently using to define the test environment, and I'm planning on bringing that back into use. That should cover this kind of thing.